### DISCUSSION TITLE
Fresh install sddm corners

### DISCUSSION THREAD

**Discussion by ntdsrd** [prasanthrangan/hyprdots#971] [Q&A]
I've installed hyprdots on arch linux following Ksk's video.

This is n times i install it in fresh. Frequently when I install and choose Corners for sddm, it go like this

![Screenshot_20240227-012631_GitHub.jpg](https://github.com/prasanthrangan/hyprdots/assets/92931547/af765460-9c56-4bce-a9d3-d05f66bc7ea1)

But today (27/02/2024), when I installed again. It show

![20240227_012751.jpg](https://github.com/prasanthrangan/hyprdots/assets/92931547/0f256b82-9fc7-45ef-95bb-e8d11214399a)

Is this the new update or I wrong in any steps?

How can I change it to Corners?

Please help me

**Comment by Dead-Slash** [comment]
i got the same probleme yesterday i have installed the hyprdots the login page was exellent but to day when i open my computer i got the same probleme as ntdsrd

**Comment by UcraNerd** [comment]
I just updated the sddm package and got the same problem...

**Reply by UcraNerd** [reply]
GUYS SOLVED!
Uninstall and install the sddm package from the official arch linux repo or downgrade the aur package to a version before 21, I think; then in `/lib/sddm/sddm.conf.d/default.conf` change `DisplayServer=x11` to `DisplayServer=wayland` then in the [THEME] selection type the name of the current theme's folder in the `Current=`, in your case `Current=Corners`. If still doesn't work, try to create a file called `sddm.conf` in `/etc` with:
> [Theme]
Current=Corners

**Comment by hoodust** [comment]
Replacing sddm-git with sddm works, but since making that change I sometimes get a black screen with cursor for a minute or so after logging in before hyprland starts. Then it will randomly log me back out once or twice after a couple minutes.

**Reply by UcraNerd** [reply]
Firstly, are you sure that the config file have wayland and not x11, if yes, have you downloaded sddm from the official arch repo? For now I didn't have any problems so I don't know. Try `sddm --test-mode --theme /usr/share/sddm/themes/Corners` and if you get error send them.

**Reply by hoodust** [reply]
Yes, changed to wayland in both default.conf and /etc/sddm.conf.d/sddm.conf. 

I installed with `yay -S sddm`.

When opening test mode the only error I see is:
```
[10:27:45.780] (II) DAEMON: Greeter errors: warning: queue 0x5592878a7150 destroyed while proxies still attached:
  wl_registry@15 still attached
```
When logging in in test mode:
```
[10:28:21.185] (II) DAEMON: Message received from greeter: Login
[10:28:21.185] (II) DAEMON: Reading from "/usr/share/wayland-sessions/hyprland.desktop"
[10:28:21.186] (EE) DAEMON: Failed to open VT master: Permission denied
[10:28:21.186] (II) DAEMON: Session "/usr/share/wayland-sessions/hyprland.desktop" selected, command: "Hyprland" for VT -1
[10:28:21.197] (II) HELPER: [PAM] Starting...
[10:28:21.197] (II) HELPER: [PAM] Authenticating...
[10:28:21.200] (II) HELPER: [PAM] Preparing to converse...
[10:28:21.200] (II) HELPER: [PAM] Conversation with 1 messages
[10:28:21.223] (II) HELPER: [PAM] returning.
[10:28:21.224] (II) DAEMON: Authentication for user  "hoo"  successful
[10:28:21.224] (WW) DAEMON: QIODevice::write (QFile, "/var/lib/sddm/state.conf"): device not open
[10:28:21.226] (II) HELPER: Starting Wayland user session: "/usr/share/sddm/scripts/wayland-session" "Hyprland"
[10:28:21.228] (II) DAEMON: Session started true
[10:28:21.228] (WW) HELPER: Failed to write utmpx:  Permission 
denied
[10:28:21.228] (II) HELPER: [PAM] Closing session
[10:28:21.228] (WW) HELPER: Failed to write utmpx:  Permission denied
[10:28:21.228] (II) HELPER: [PAM] Ended.
[10:28:21.229] (WW) DAEMON: Auth: sddm-helper exited with 3
[10:28:21.229] (II) DAEMON: Greeter stopping...
[10:28:21.244] (II) DAEMON: Greeter errors: warning: queue 0x5592878b7470 destroyed while proxies still attached:
  zwp_linux_dmabuf_feedback_v1@40 still attached
  zwp_linux_dmabuf_feedback_v1@32 still attached
  zwp_linux_dmabuf_feedback_v1@30 still attached
  zwp_linux_dmabuf_feedback_v1@21 still attached
  zwp_linux_dmabuf_v1@22 still attached
```

**Reply by UcraNerd** [reply]
Do you have Nvidia?

**Reply by hoodust** [reply]
Yes

**Reply by UcraNerd** [reply]
Can you send `/etc/pam.d/sddm` and `/etc/pam.d/sddm-autologin`?
And if you have, `/etc/modules-load.d/nvidia-drm.conf`

**Reply by hoodust** [reply]
pam.d/sddm
```
#%PAM-1.0

auth        include     system-login
-auth       optional    pam_gnome_keyring.so
-auth       optional    pam_kwallet5.so

account     include     system-login

password    include     system-login
-password   optional    pam_gnome_keyring.so    use_authtok

session     optional    pam_keyinit.so          force revoke
session     include     system-login
-session    optional    pam_gnome_keyring.so    auto_start
-session    optional    pam_kwallet5.so         auto_start
```
sddm-autologin
```
#%PAM-1.0
auth        required    pam_env.so
auth        required    pam_faillock.so preauth
auth        required    pam_shells.so
auth        required    pam_nologin.so
auth        required    pam_permit.so
-auth       optional    pam_gnome_keyring.so
-auth       optional    pam_kwallet5.so
account     include     system-local-login
password    include     system-local-login
session     include     system-local-login
-session    optional    pam_gnome_keyring.so auto_start
-session    optional    pam_kwallet5.so auto_start
```
and in /etc/modules-load.d/ I just have `nvidia-dev-dkms-tkg.conf` (I use dev vulkan drivers from the TKG script) with the single line in it: `nvidia-uvm`

**Reply by hoodust** [reply]
I grep'd journalctl for sddm and saw something about the display server failing and failing over to x11-user and switching from VT 1 to 2 to 1 to 3... so I just set `DisplayServer=x11-user` in the conf and it seems to work just fine again.

**Reply by hoodust** [reply]
Aw, spoke too soon. After rebooting it happened again. Logged me out THREE times.
```
❯ journalctl -b | grep -i "Jumping to VT "
Feb 28 16:58:06 {hostname} sddm-helper[833]: Jumping to VT 1
Feb 28 16:58:23 {hostname} sddm-helper[925]: Jumping to VT 2
Feb 28 16:59:54 {hostname} sddm-helper[1973]: Jumping to VT 1
Feb 28 17:00:01 {hostname} sddm-helper[2033]: Jumping to VT 3
Feb 28 17:01:31 {hostname} sddm-helper[3317]: Jumping to VT 1
Feb 28 17:01:39 {hostname} sddm-helper[3341]: Jumping to VT 4
Feb 28 17:03:10 {hostname} sddm-helper[5326]: Jumping to VT 1
Feb 28 17:03:16 {hostname} sddm-helper[5348]: Jumping to VT 5
```
Here's the full journalctl of all sddm messages: https://hastebin.com/share/umilisexik.swift

**Reply by UcraNerd** [reply]
Can it be this?
Black screen after logout with NVIDIA graphics card
One may encounter a complete black screen or with only cursor/display device logo on it after the logout of any user. This happens because sddm.service starts faster than the NVIDIA drivers. Consider using [early KMS](https://wiki.archlinux.org/title/NVIDIA#Early_loading)

**Reply by hoodust** [reply]
I don't think so because I have all those modules in my mkinitcpio.conf already. Plus it's not when logging out, it's when logging in. Never had this problem before switching from sddm-git to sddm. Thanks though

**Reply by UcraNerd** [reply]
Sorry, I'm not that of an expert to find a solution, i think it's an nvidia only problem because me and others, that don't have nvidia, don't have this issue. Maybe there are some nvidia linux community forums, so try to ask to someone who might have experienced this problem before, good luck.

**Reply by hoodust** [reply]
Fyi: I think I fixed this. There was another config file that I renamed .backup or whatever and I think it was conflicting, even though the extension wasn't .conf. Since I deleted it (leaving just ONE file in /etc/sddm.conf.d/ ) I haven't had any issues.

**Comment by CrownKingClown** [comment]
broke my head because on a fresh install today i've got: `"The theme at "/usr/share/sddm/themes/breeze" requires missing "/usr/bin/sddm-greeter" . Using fallback theme.` The login screen was literally a mess

after hours of cleaning caches, reinstalling, linking, fixed with.. 
`sudo apt install sddm-theme-breeze`
